Example/wifi_sta/firmware/src/app.c

```diff
@@ -75,7 +114,176 @@ uint8_t CACHE_ALIGN work[SYS_FS_FAT_MAX_SS];
 
 
 /* TODO:  Add any necessary local functions.
-*/
+ */
+
+
+bool validateUserPass (char * user, char * pass){
+    
+    if ((strlen(user) > 10) || (strlen(pass) > 10)) {
+        
+        SYS_CONSOLE_MESSAGE("Username or Password is too long\n\r");
+        return false;
+    }
+
+    return true;
+}
+
+bool FTP_Command_Initialize(void){
+    
+    if (!SYS_CMD_ADDGRP(ftpCmdTbl, sizeof(ftpCmdTbl)/sizeof(*ftpCmdTbl), "ftp", ": commands"))
+    {
+        SYS_ERROR(SYS_ERROR_ERROR, "Failed to create FTP Commands\r\n");
+        return false;
+    }
+}
+
+static inline void _Command_setECC(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    setECC = true;                          //ECC will be used to store FTP server login credentials
+    appData.state = APP_ECC_Init;
+}
+
+static int _Command_AddUser(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    if(setECC){
+        clientLoginInfo client;
+        
+        strncpy(client.username, argv[1], 10);
+        strncpy(client.password, argv[2], 10);
+        if(validateUserPass(client.username, client.password))
+        {
+            status = atcab_write_zone(ECC_TNG_DATA_ZONE, ECC_TNG_GP_DATA_SLOT, ECC_TNG_GP_DATA_SLOT_BLOCK_BASE, 0, (uint8_t*)&client, 32);
+            if (ATCA_SUCCESS == status) 
+            {
+                SYS_CONSOLE_PRINT("Successfully registered User: \"%s\"\r\n", client.username);
+                appData.state = APP_STATE_SERVICE_TASKS;    
+            }
+        }
+    }
+    else{
+        SYS_CONSOLE_MESSAGE("Not in ECC mode. Please use \"setecc\" command to enable ECC mode\n\r");
+    }
+}
+
+static int _Command_RemoveUser(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    if(setECC)
+    {
+#define     resetDefault        "FFFFFFFF"    
+        clientLoginInfo client1;
+        
+        strncpy(client1.username, argv[1], 10);
+        strncpy(client1.password, argv[2], 10);
+        
+        if(validateUserPass(client1.username, client1.password))
+        {
+            char tempRead[32]; 
+            clientLoginInfo *tempRead2;
+            tempRead2 = (clientLoginInfo *)tempRead;
+            
+            status = atcab_read_zone(ECC_TNG_DATA_ZONE, ECC_TNG_GP_DATA_SLOT, ECC_TNG_GP_DATA_SLOT_BLOCK_BASE, 0, tempRead, 32);
+            if (ATCA_SUCCESS == status) 
+            {
+                if(0 == strncmp(client1.username, tempRead2->username, sizeof(tempRead2->username)))
+                {
+                    atcab_write_zone(ECC_TNG_DATA_ZONE, ECC_TNG_GP_DATA_SLOT, ECC_TNG_GP_DATA_SLOT_BLOCK_BASE, 0, (uint8_t*)resetDefault, 32);
```

Comment:
user default writes resetDefault string into the ECC which is nothing but having an user with user name 'FFFFFFFF' and the older password

The right way to make use of ECC space is to have a magic number (unique bytes) to determine if the details are valid or not. To make it invalid write a invalid magic number

Reply:
removed the function